week08/movie/src/App.js

```diff
@@ -1,15 +1,34 @@
 import React, { useEffect } from "react";
-import axios from "axios";
 import MainPage from "./pages/MainPage";
 import "./App.css";
+import { Provider } from "react-redux";
+import store from "./store";
+import { ThemeProvider } from "styled-components";
+import { useSelector } from "react-redux";
 
 function App() {
+  const theme = useSelector((state) => state.themeSlicer.theme);
+
+  useEffect(() => {
+    const body = document.body;
+    body.style.backgroundColor = theme.mode === "dark" ? "#111" : "#ffd464";
+    body.style.backgroundImage =
+      theme.mode === "light"
+        ? "linear-gradient(0deg, rgba(17, 17, 17, 1) 0%, rgba(34, 34, 34, 1) 45%, rgba(84, 84, 84, 1) 100%)"
+        : "linear-gradient(0deg, rgba(255,212,100,1) 0%, rgba(251,231,185,1) 25%, rgba(255,255,255,1) 72%)";
+
+    body.style.backgroundRepeat = "no-repeat";
+    body.style.backgroundAttachment = "fixed";
+  }, [theme]);
+
   return (
-    <body>
-      <div className="main">
-        <MainPage />
-      </div>
-    </body>
+    <Provider store={store}>
+      <ThemeProvider theme={theme}>
+        <div className="main">
+          <MainPage />
+        </div>
+      </ThemeProvider>
+    </Provider>
```

Comment:
이미 index.js에서 `<App>` 컴포넌트 상위에 `<Provider store={store}>` 컴포넌트로 store를 공급해주고 있습니다.
따라서 App 컴포넌트에서 중복으로 다시 store을 공급해줄 필요가 없기 때문에 해당 부분을 삭제하는 것이 좋습니다.

```suggestion
    <ThemeProvider theme={theme}>
      <MainPage />
    </ThemeProvider>
```